spec/features/create_anschutz_work_spec.rb

```diff
@@ -1,20 +1,102 @@
+    )
+    login_as user
+  end
+
+  it 'renders the new Anschutz Work page' do
+    visit "concern/#{work_type.to_s.pluralize}/new"
+    expect(page).to have_content "Add New Anschutz Work"
+  end
+
+  it 'adds files to work' do
+    visit "concern/#{work_type.to_s.pluralize}/new"
+    click_link "Files" # switch tab
+    expect(page).to have_content "Add files"
+    expect(page).to have_content "Add folder"
+    within('span#addfiles') do
+      attach_file("files[]", Rails.root.join('spec', 'fixtures', 'hyrax', 'image.jp2'), visible: false)
+      attach_file("files[]", Rails.root.join('spec', 'fixtures', 'hyrax', 'jp2_fits.xml'), visible: false)
+    end
+  end
+
+  it 'applys work visibility' do
+    visit "concern/#{work_type.to_s.pluralize}/new"
+    visibility = :open
+    find('body').click
+    choose("#{work_type}_visibility_#{visibility}")
+    case visibility
+    when :open
+      expect(page).to have_content('Please note, making something visible to the world (i.e. marking this as Public) may be viewed as publishing which could impact your ability to')
+    end
+  end
+
+  it 'saves the work' do
+    visit "concern/#{work_type.to_s.pluralize}/new"
+    click_link "Descriptions" # switch tab
+    fill_in("#{work_type}_title", with: 'My Test Work')
+    select('Organisational', from: "#{work_type}_creator__creator_name_type")
+    fill_in("#{work_type}_creator__creator_organization_name", with: 'Ubiquity Press')
+    check('agreement')
+    click_on('Save')
+    expect(page).to have_content('My Test Work')
+    expect(page).to have_content('Public') # This test is failing and works are not being saved as public, need to look into why
+    expect(page).to have_content("Your files are being processed by Hyku in the background.")
+  end
+
+  context "when rendering the form" do
+    before do
+      visit "concern/#{work_type.to_s.pluralize}/new"
+      click_on "Additional fields"
+    end
+
+    it "renders all simple worktype fields" do
+      worktype_simple_fields = %w[title alt_title date_published_text abstract resource_type license
+                                  place_of_publication language subject_text mesh add_info advisor publisher source
+                                  journal_frequency funding_description citation table_of_contents
+                                  references extent medium library_of_congress_classification committee_member
+                                  time part_of rights_statement qualification_subject_text qualification_grantor qualification_level
+                                  qualification_name is_format_of source_identifier]
+      worktype_simple_fields.each do |field|
+        expect(page).to have_field("#{work_type}_#{field}")
+      end
+    end
+
+    it "renders all complex fields" do
+      worktype_complex_fields = %w[creator]
+      worktype_complex_fields.each do |field|
+        expect(page).to have_field("#{work_type}_#{field}__#{field}_family_name")
```

Comment:
We're creating the field name manually a lot here, can we replace the `"#{work_type}_#{field}__#{field}"` with a `let(:field_name)` at the top?

This example could then be: 

```expect(page).to have_field("#{field_name}_family_name")```

Reply:
I don't think so because field is undefined outside the loop

Reply:
For the loops you are probably right, but for the others we could.